dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,658 @@
+    .. [1] "Hyperband: A novel bandit-based approach to hyperparameter
+           optimization", 2016 by L. Li, K. Jamieson, G. DeSalvo, A.
+           Rostamizadeh, and A. Talwalkar.  https://arxiv.org/abs/1603.06560
+    .. [2] "Massively Parallel Hyperparameter Tuning", 2018 by L. Li, K.
+            Jamieson, A. Rostamizadeh, K. Gonina, M. Hardt, B. Recht, A.
+            Talwalkar.  https://openreview.net/forum?id=S1Y7OOlRZ
+
+    """
+
+    def __init__(
+        self,
+        model,
+        params,
+        max_iter=81,
+        eta=3,
+        asynchronous=True,
+        random_state=42,
+        scoring=None,
+        test_size=0.15,
+    ):
+        self.model = model
+        self.params = params
+        self.max_iter = max_iter
+        self.eta = eta
+        self.test_size = test_size
+        self.random_state = random_state
+        self.asynchronous = asynchronous
+
+        self.best_score = None
+        self.best_params = None
+
+        super(HyperbandCV, self).__init__(model, scoring=scoring)
+
+    def fit(self, X, y, **fit_params):
+        """Find the best parameters for a particular model
+
+        Parameters
+        ----------
+        X, y : array-like
+        **fit_params
+            Additional partial fit keyword arguments for the estimator.
+        """
+        return default_client().sync(self._fit, X, y, **fit_params)
+
+    @gen.coroutine
+    def _fit(self, X, y, **fit_params):
+        if isinstance(X, np.ndarray):
+            X = da.from_array(X, chunks=X.shape)
+        if isinstance(y, np.ndarray):
+            y = da.from_array(y, chunks=y.shape)
+
+        # We always want a concrete scorer, so return_dask_score=False
+        # We want this because we're always scoring NumPy arrays
+        self.scorer_ = check_scoring(self.model, scoring=self.scoring)
+        self.best_score = -np.inf
+        r = yield _hyperband(
+            self.model,
+            self.params,
+            X,
+            y,
+            max_iter=self.max_iter,
+            fit_params=fit_params,
+            eta=self.eta,
+            random_state=self.random_state,
+            test_size=self.test_size,
+            scorer=self.scorer_,
+            asynchronous=self.asynchronous,
+        )
+        params, model_meta_futures, history, meta = r
+
+        self.meta_ = meta
+        self.history_ = history
+
+        cv_res, best_idx = _get_cv_results(meta, params)
+        best_id = cv_res["model_id"][best_idx]
+
+        best_model_and_meta = yield model_meta_futures[best_id]
+        self.best_estimator_ = best_model_and_meta[0]
+
+        self.cv_results_ = cv_res
+        self.best_index_ = best_idx
+        self.n_splits_ = 1  # TODO: increase this! It's hard-coded right now
+        self.multimetric_ = False
+
+        raise gen.Return(self)
+
+    def fit_metadata(self, meta=None):
+        """Get information about how much computation is required for
+        :func:`~dask_ml.model_selection.HyperbandCV.fit`.
```

Comment:
Note whether `self` has to be fitted first or not.

Reply:
`self` does not have to be fitted to use `metadata` (which is documented)